src/lib/mdns/Resolver.h

```diff
@@ -35,7 +35,50 @@ struct ResolvedNodeData
     Inet::IPAddress mAddress;
     uint16_t mPort;
 };
+struct CommissionableNodeData
+{
+    // TODO(cecille): is 4 OK? IPv6 LL, GUA, ULA, IPv4?
+    static constexpr int kMaxIPAddresses = 5;
+    // Largest host name is 64-bits in hex.
+    static constexpr int kHostNameSize = 16;
+
+    char hostName[kHostNameSize + 1];
+    uint16_t longDiscriminator;
+    uint16_t vendorId;
+    uint16_t productId;
+    int numIPs;
+    Inet::IPAddress ipAddress[kMaxIPAddresses];
+    void Reset()
+    {
+        memset(hostName, 0, kHostNameSize + 1);
+        longDiscriminator = 0;
+        vendorId          = 0;
+        productId         = 0;
+        numIPs            = 0;
+        for (int i = 0; i < kMaxIPAddresses; ++i)
+        {
+            ipAddress[i] = chip::Inet::IPAddress::Any;
+        }
+    }
+    CommissionableNodeData() { Reset(); }
+    bool IsHost(const char * host) const { return strncmp(host, hostName, kHostNameSize) == 0; }
+    bool IsValid() const { return !IsHost("") && ipAddress[0] != chip::Inet::IPAddress::Any; }
+}; // namespace Mdns
```

Comment:
is this a stray comment? I think this may be the end of 'CommisionableNodeData' instead of namespace